Sources/PackageModel/ToolchainConfiguration.swift

```diff
@@ -96,21 +96,80 @@ extension ToolchainConfiguration {
         public var manifestAPI: AbsolutePath
         public var pluginAPI: AbsolutePath
 
-        public init(manifestAPI: AbsolutePath, pluginAPI: AbsolutePath) {
+        public init(manifestAPI: AbsolutePath, manifestMinimumDeploymentTarget: PlatformVersion? = nil, pluginAPI: AbsolutePath, pluginMinimumDeploymentTarget: PlatformVersion? = nil) {
+            #if os(macOS)
+            if let manifestMinimumDeploymentTarget = manifestMinimumDeploymentTarget {
+                self.manifestMinimumDeploymentTarget = manifestMinimumDeploymentTarget
+            } else if let manifestMinimumDeploymentTarget = try? MinimumDeploymentTarget.computeMinimumDeploymentTarget(of: Self.macOSPackageDescriptionPath(for: manifestAPI), platform: .macOS) {
+                self.manifestMinimumDeploymentTarget = manifestMinimumDeploymentTarget
+            } else {
+                self.manifestMinimumDeploymentTarget = Self.defaultMinimumDeploymentTarget
+            }
+
+            if let pluginMinimumDeploymentTarget = pluginMinimumDeploymentTarget {
+                self.pluginMinimumDeploymentTarget = pluginMinimumDeploymentTarget
+            } else if let pluginMinimumDeploymentTarget = try? MinimumDeploymentTarget.computeMinimumDeploymentTarget(of: Self.macOSPluginLibraryPath(for: pluginAPI), platform: .macOS) {
+                self.pluginMinimumDeploymentTarget = pluginMinimumDeploymentTarget
+            } else {
+                self.pluginMinimumDeploymentTarget = Self.defaultMinimumDeploymentTarget
+            }
+            #else
+            precondition(manifestAPIMinimumDeploymentTarget == nil && pluginAPIMinimumDeploymentTarget == nil, "deployment targets can only be specified on macOS")
```

Comment:
I considered platform-specific signatures for the initializers instead off this, but that doesn't seem to really be possible with `#if` in a way that doesn't mean essentially having two copies of them, one for macOS and one for other platforms. That does seem a bit much, so I instead opted for this precondition.